images/testing.go

```diff
@@ -23,6 +28,46 @@ func (f *FakeProvider) GetImage(_ context.Context, token string) (*Image, error)
 	return nil, ErrImageNotFound
 }
 
+// GetImageURL returns the URL of the image associated with a given alert.
+func (f *FakeProvider) GetImageURL(_ context.Context, alert types.Alert) (string, error) {
+	uri, err := getImageURI(alert)
+	if err != nil {
+		return "", err
+	}
+
+	for _, img := range f.Images {
+		if img.Token == uri || img.URL == uri {
+			return img.URL, nil
+		}
+	}
+	return "", ErrImageNotFound
+}
+
+// GetRawImage returns an io.Reader to read the bytes of the image associated with a given alert.
+func (f *FakeProvider) GetRawImage(_ context.Context, alert types.Alert) (io.Reader, error) {
+	uri, err := getImageURI(alert)
+	if err != nil {
+		return nil, err
+	}
+
+	uriString := string(uri)
+	for _, img := range f.Images {
+		if img.Token == uriString || img.URL == uriString {
+			return strings.NewReader("test"), nil
+		}
+	}
+	return nil, ErrImageNotFound
+}
+
+// getImageURI is a helper function to retrieve the image URI from the alert annotations as a string.
+func getImageURI(alert types.Alert) (string, error) {
+	uri, ok := alert.Annotations[models.ImageTokenAnnotation]
+	if !ok {
+		return "", fmt.Errorf("no image uri in annotations")
```

Comment:
I think you should make this a named error. I suggest `ErrNoImageURI`

Because then on the callsite, you can do things like 

```golang
	uri, err := getImageURI(alert)
        if err != nil {
	  if errors.Is(ErrNoImageURI) {
		  return nil, ErrImageNotFound
	  }
         return err
}
```